src/main/java/org/broadinstitute/dsm/db/structure/DBElement.java

```diff
@@ -9,11 +14,89 @@
     public String tableAlias;
     public String primaryKey;
     public String columnName;
+    public SqlDateConverter dateConverter;
 
     public DBElement(String tableName, String tableAlias, String primaryKey, String columnName) {
         this.tableName = tableName;
         this.tableAlias = tableAlias;
         this.primaryKey = primaryKey;
         this.columnName = columnName;
     }
+
+    public DBElement(String tableName, String tableAlias, String primaryKey, String columnName, SqlDateConverter dateConverter) {
+        this(tableName, tableAlias, primaryKey, columnName);
+        this.dateConverter = dateConverter;
+    }
+
+    /**
+     * Use this for dates that are stored as millis since the epoch.
+     */
+    public static class EpochDateConverter implements DateConverter {
+
+        public String convertArgToSql(Instant arg) {
+            return Long.toString(arg.toEpochMilli());
+        }
+
+        public String convertColumnForSql(String column) {
+            return column;
+        }
+
+        public String convertArgToSqlForDay(Instant arg) {
+            return "DATE(FROM_UNIXTIME(" + arg.toEpochMilli()/1000 + "))";
+        }
+
+        public String convertColumnForSqlForDay(String column) {
+            return "DATE(FROM_UNIXTIME(" + column + "/1000))";
+        }
+    }
+
+    /**
+     * Use this for simple yyyy-mm-dd string columns
+     */
+    public static class StringDayConverter implements DateConverter {
+
+        public String convertArgToSql(Instant arg) {
+            return "STR_TO_DATE(" + "'" + DateTimeFormatter.ofPattern("yyyy-MM-dd").withZone(ZoneId.of("UTC")).format(arg) + "'" + ",'%Y-%m-%d')";
```

Comment:
maybe using already existing `DateTimeFormatter.ISO_LOCAL_DATE` format could be more cleaner?